framework/src/org/checkerframework/framework/flow/CFAbstractTransfer.java

```diff
@@ -785,6 +822,18 @@ protected void processCommonAssignment(TransferInput<V, S> in, Node lhs,
         // add new information based on conditional postcondition
         processConditionalPostconditions(n, method, tree, thenStore, elseStore);
 
+        if (inferSignatures) {
+            // Finds the receiver's type
+            Tree receiverTree = n.getTarget().getReceiver().getTree();
+            if (receiverTree == null) {
+                // If there is no receiver, then get the class being visited.
+                receiverTree = analysis.getContainingClass(n.getTree());
```

Comment:
In the first assignment to receiverTree, it will be null if the method's receiver is "this". If that is the case, the second assignment retrieves the class corresponding to "this". But if that is called from a static context, the second assignment will also return null. So I'm deferring the handling of a call in a static context to SignatureInferenceScenes.updateInferredMethodParametersTypes. Does that makes sense to you? I'll add a comment about the fact that when "this" is the receiver, null is returned.